src/osd/OSD.cc

```diff
@@ -3882,13 +3882,12 @@ void OSD::maybe_update_heartbeat_peers()
       PG *pg = i->second;
       pg->heartbeat_peer_lock.Lock();
       dout(20) << i->first << " heartbeat_peers " << pg->heartbeat_peers << dendl;
-      for (set<int>::iterator p = pg->heartbeat_peers.begin();
-	   p != pg->heartbeat_peers.end();
-	   ++p)
-	if (osdmap->is_up(*p))
-	  _add_heartbeat_peer(*p);
-      for (set<int>::iterator p = pg->probe_targets.begin();
-	   p != pg->probe_targets.end();
+      set<int> heartbeat_peers;
```

Comment:
could you elaborate a little bit, why it is a better way in the commit message? "better" is too vague a word, imo.